zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -267,7 +267,7 @@ object Endpoint {
    * Constructs an endpoint for an HTTP DELETE method, whose path is described
    * by the specified path codec.
    */
-  def delete[Input](path: PathCodec[Input]): Endpoint[Input, ZNothing, ZNothing, EndpointMiddleware.None] = {
+  def delete[Input](path: PathQueryCodec[Input]): Endpoint[Input, ZNothing, ZNothing, EndpointMiddleware.None] = {
```

Comment:
Rather change all of these to: `HttpCodec[HttpCodecType.Query with HttpCodecType.Path, ...]`

Or add a type alias: `type PathQueryCodec[A] = HttpCodec[HttpCodecType.Query with HttpCodecType.Path, ...]`